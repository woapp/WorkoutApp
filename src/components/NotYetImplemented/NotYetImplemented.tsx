import React, { FunctionComponent } from 'react';
import { View, Text, Image } from 'react-native';
import styled from '@woap/utils/styled-components';

import images from '../../assets/images';

type INotYetImplemented = {
  pageTitle: string;
};

const ILLUSTRATION_WIDTH = '70%';
const ILLUSTRATION_HEIGHT = '50%';

export const NotYetImplemented: FunctionComponent<INotYetImplemented> = ({ pageTitle }) => {
  return (
    <Container>
      <Illustration source={images.workInProgress} resizeMode="contain" />
      <Text>We are currently working on the page {pageTitle}</Text>
    </Container>
  );
};

const Container = styled(View)`
  flex-grow: 1;
  justify-content: center;
  align-items: center;
  background-color: white;
`;

const Illustration = styled(Image)`
  align-self: center;
  height: ${ILLUSTRATION_HEIGHT};
  width: ${ILLUSTRATION_WIDTH};
`;
