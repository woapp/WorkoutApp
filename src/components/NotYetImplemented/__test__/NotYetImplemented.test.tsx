import React from 'react';
import { render } from 'react-native-testing-library';

import { NotYetImplemented } from '../NotYetImplemented';

describe('[Component] NotYetImplemented', () => {
  it('should render correctly', () => {
    const component = render(<NotYetImplemented />);
    expect(component).toMatchSnapshot();
  });
});