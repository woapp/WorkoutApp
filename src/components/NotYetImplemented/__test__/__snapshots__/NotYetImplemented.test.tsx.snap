// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] NotYetImplemented should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexGrow": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/theme/images/workInProgress/workInProgress.png",
      }
    }
    style={
      Array [
        Object {
          "alignSelf": "center",
          "height": "50%",
          "width": "70%",
        },
      ]
    }
  />
  <Text>
    We are currently working on the page 
  </Text>
</View>
`;
